### DISCUSSION TITLE
Status & Body regex test or match supported

### DISCUSSION THREAD

**Discussion by scottstark** [mockoon/mockoon#1592] [Help]
I have a mocked 400 response with several Rules applied to the request body. I want to then catch which rules failed in my Status & Body in order to return the correct error messages. Is there a way to call regex test or match in the Status & Body using helpers or handlebars?

If I have a couple rules like these:
`Body, id, !, regex, ^.{7,10}$`
`Body, name, !, regex, ^.{1,50}$`

I want to do something like this in the 'Status & Body' but make use of regex:

```handlebars
{{#unless (bodyRaw 'id')}}
  {
    "code": "400001",
    "message": "Field $.id is invalid or missing",
    "field": "$.id",
    "location": "BODY"
  }
{{else}}
  {{#if (gt (len (bodyRaw 'id')) 10)}}
    {
      "code": "400001",
      "message": "Field $.id is invalid or missing",
      "field": "$.id",
      "location": "BODY"
    }
  {{/if}}
  {{#if (lt (len (bodyRaw 'id')) 7)}}
    {
      "code": "400001",
      "message": "Field $.id is invalid or missing",
      "field": "$.id",
      "location": "BODY"
    }
  {{/if}}
{{/unless}}


{{#unless (bodyRaw 'name')}}
  {
    "code": "400001",
    "message": "Field $.name is invalid or missing",
    "field": "$.name",
    "location": "BODY"
  }
{{else}}
  {{#if (gt (len (bodyRaw 'name')) 50)}}
    {
      "code": "400001",
      "message": "Field $.name is invalid or missing",
      "field": "$.name",
      "location": "BODY"
    }
  {{/if}}
{{/unless}}
```

**Comment by 255kb** [comment]
You usually want to create **multiple** responses with rules, or deal with this kind of validation in the template. But templates do not communicate with the rules (but maybe I misunderstood what you are trying to do). 

A typical setup is usually the following: 
- default response (400, etc.) with no rules
- 200 response with one or more rules
- and so on

I suggest you read the [docs](https://mockoon.com/docs/latest/route-responses/dynamic-rules/) and this [tutorial](https://mockoon.com/tutorials/responses-and-rules/) to understand how rules work.

You can perfectly have one response and customize it with a template, but it's probably a bit more complicated. Also, considering there is no regex helper.

In v9 [JSON schema rules](https://mockoon.com/docs/latest/route-responses/dynamic-rules/#4-comparison-operator) were introduced (some improvements are coming to v9.1.0 soon), which makes this kind of validation way easier.
